front-end/src/app/shared/models/transaction.model.ts

```diff
@@ -0,0 +1,46 @@
+import { BaseModel } from './base.model';
+import { Contact } from './contact.model';
+import { MemoText } from './memo-text.model';
+
+export abstract class Transaction extends BaseModel {
+  id: string | undefined;
+
+  // FECFile spec properties
+
+  form_type: string | undefined;
+  filer_committee_id_number: string | undefined;
+  transaction_id: string | null = null; // This is a required field and must exist
+
+  // FECFile Online custom properties
+
+  transaction_type_identifier: string | undefined;
+  itemized: boolean | undefined;
+
+  parent_transaction: Transaction | undefined;
+  parent_transaction_id: string | undefined; // Foreign key to the parent transaction db record
```

Comment:
`parent_transaction_id` had to change to `parent_transaction_object_id` for the ContentType workaround to work.  I'm seeing if there's a way to get around that

Reply:
This change will have to wait until the API changes for sched B are merged into the develop branch. Will stay as is for this PR.